agent/dehydrate.js

```diff
@@ -40,6 +40,13 @@ function dehydrate(data: Object, cleaned: Array<Array<string>>, path?: Array<str
       type: 'function',
     };
   }
+  //check if data is of type date. typeof doesn't support Date, so instanceof is being used
+  if (data instanceof Date) {
```

Comment:
I'm not sure this will work across iframes. Let's use `data != null && typeof data == 'object' && Object.prototype.toString.call(data) === '[object Date]'`. See explanation [here](http://stackoverflow.com/questions/643782/how-to-check-whether-an-object-is-a-date#comment2644531_643827). This is what [Lodash does](https://github.com/lodash/lodash/blob/master/isDate.js), too.